File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int l = 0, r = 1, max_profit = 0;
        while (r < prices.Length) {
            if (prices[l] < prices[r]) {
                int profit = prices[r]-prices[l];
                max_profit = Math.Max(profit, max_profit);
            } else {
                l = r;
            }
            r++;
        }
        return max_profit;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-3.ts
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights: number[]): number {
        let l = 0, r = heights.length-1, max_height = 0;
        while (l < r ) {
            const height: number = (r-l) * Math.min(heights[l], heights[r]);
            max_height = Math.max(max_height, height); 
            if (heights[l] < heights[r]) {
                l++;
            } else {
                r--;
            }
        }
        return max_height;
    }
}
